src/java/org/apache/cassandra/tools/nodetool/Status.java

```diff
@@ -149,28 +150,33 @@ private void addNodesHeader(boolean hasEffectiveOwns, TableBuilder tableBuilder)
     private void addNode(String endpoint, Float owns, String epDns, String token, int size, boolean hasEffectiveOwns,
```

Comment:
What would you think of just refactoring this to accept the HostStatWithPort? Then we have access to `ipOrDns()` as well as the `endpoint`, and we don't need to resolve again.
https://github.com/apache/cassandra/blob/401e933b7395892bf0356f88308f64b94be84601/src/java/org/apache/cassandra/tools/nodetool/HostStat.java#L24)